hardware/DomoticzTCP.cpp

```diff
@@ -127,6 +127,7 @@ bool DomoticzTCP::StopHardware()
 	if (m_thread)
 	{
 		RequestStop();
+		disconnectTCP();
```

Comment:
This should actually be done in the worker thread once the request for stop is received at the end of the worker

Reply:
Problem is that the worker thread is hanging on read() before it receives any stop requests.. By calling disconnectTCP before it will send a shutdown / interrupt to wake it up. That's why it's not working when putting it at the end of the worker thread.